src/components/selectable/selectable_list/selectable_list_item.tsx

```diff
@@ -164,19 +170,38 @@ export class EuiSelectableListItem extends Component<
       instruction = (
         <EuiI18n
           token="euiSelectableListItem.excludedOptionInstructions"
-          default="To uncheck this option, press enter."
+          default="To uncheck this option, press Enter."
+        />
+      );
+    } else if (allowExclusions && checked === 'mixed') {
+      state = (
+        <EuiI18n
+          token="euiSelectableListItem.excludedOption"
+          default="Mixed (indeterminate) option."
+        />
+      );
+      instruction = (
+        <EuiI18n
+          token="euiSelectableListItem.unckeckedOptionInstructions"
+          default="To check this option for all, press Enter once. To exclude this option for all, press Enter twice."
         />
       );
     } else if (allowExclusions && !checked) {
       instruction = (
         <EuiI18n
           token="euiSelectableListItem.unckeckedOptionInstructions"
-          default="To select this option, press enter."
+          default="To check this option, press Enter."
         />
       );
     }
 
-    const isChecked = !disabled && typeof checked === 'string';
+    let isChecked: boolean | EuiSelectableOptionCheckedType =
+      !disabled && typeof checked === 'string';
+    // 'Mixed' has its own logic because it is the only aria-checked
+    // state we can't derive with the `isChecked` boolean alone.
+    if (isChecked && checked === 'mixed') {
+      isChecked = 'mixed';
+    }
```

Comment:
the fact that we had to write a two line comment explaining what is going on with this is probably a code smell / indicates that it could be written more cleanly. IMO, we can use `useMemo`'s function API for an early return and a switch to make this code clearer.

```suggestion
    const isChecked: boolean | EuiSelectableOptionCheckedType = useMemo(() => {
      if (disabled) return undefined;
      switch (checked) {
        case 'on':
        case 'off':
          return true;
        case 'mixed':
          return 'mixed';
        case undefined:
        default:
          return false;
      }
    }, [checked]);
```

Although... on second thought and after looking at https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-checked#values more closely, I think we have a problem here 😬 Let's chat about this synchronously when you get a chance

Reply:
This `useMemo()` is a lot cleaner. I'll refactor and follow up with you RE the `aria-checked="mixed"` to make sure we're in agreement.

Reply:
Returning back to GH to close the loop on this - Trevor has tested w/ both axe and various screen readers and the `aria-checked="mixed" role="option"` combo still surprisingly works despite it not being "supported".

We're moving ahead with it, although I do think we should leave a comment above the mixed case noting that we made this decision despite knowing MDN says not to, and to keep an eye on this for the future.